src/main/java/racing/domain/RacingCars.java

```diff
@@ -0,0 +1,45 @@
+package racing.domain;
+
+import racing.rule.MoveRule;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class RacingCars {
+
+  private final List<RacingCar> racingCars;
+
+  private RacingCars(int totalPlayer) {
+    this(IntStream.range(0, totalPlayer)
+            .mapToObj(RacingCar::create)
+            .collect(Collectors.toList()));
+  }
+
+  private RacingCars(List<RacingCar> racingCars) {
+    this.racingCars = racingCars;
+  }
+
+  public static RacingCars create(int totalPlayer) {
+    return new RacingCars(totalPlayer);
+  }
+
+  public static RacingCars copyFrom(RacingCars racingCars) {
+    List<RacingCar> result = new ArrayList<>();
+    for (RacingCar car : racingCars.getRacingCars()) {
+      result.add(RacingCar.copyFrom(car));
+    }
+    return new RacingCars(result);
+  }
+
+  public void race(MoveRule moveRule) {
+    for (RacingCar car : racingCars) {
+      car.move(moveRule);
+    }
+  }
+
+  public List<RacingCar> getRacingCars() {
+    return this.racingCars;
+  }
```

Comment:
일급 컬렉션 잘 적용해주셨네요 👍 💯 
`Collections.unmodifiablelist()`를 활용하여 외부에서 컬렉션을 건드리지 못하도록 `read-only`로 리턴하면 좋을 것 같아요!

```suggestion
  public List<RacingCar> getRacingCars() {
    return Collections.unmodifiableList(racingCars);
  }
```

Reply:
안녕하세요 경준님 ~

getRacingCars에서 UnmodifiableList 타입의 리스트를 만드는것도 좋겠지만, 
외부에서 주입된 원본 객체의 주소를 참조하고 있음으로 완벽한 read-only 객체라고 볼 수 없을거 같습니다.

따라서 방어적 복사 기법을 통해 외부에서 넘겨준 객체를 끊고 새로운 복사본 객체를 생성해서 UnmodifiableList 를 감싸면 진정한 read-only라 생각하는데 경준님은 어떻게 생각하시나요 ??


``` java
private RacingCars(List<RacingCar> racingCars) {
    this.racingCars = Collections.unmodifiableList(new ArrayList<>(racingCars));
}
```